app/src/androidTest/java/org/mozilla/reference/browser/FirefoxTestRunner.kt

```diff
@@ -0,0 +1,16 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.reference.browser
+
+import android.app.Application
+import android.content.Context
+import androidx.test.runner.AndroidJUnitRunner
+
+class FirefoxTestRunner : AndroidJUnitRunner() {
+
+    override fun newApplication(cl: ClassLoader?, className: String?, context: Context?): Application {
+        return super.newApplication(cl, FirefoxTestApplication::class.java.name, context)
+    }
```

Comment:
Why are we overriding this here? Won't we need the actual application object for a bunch of tests?

Reply:
We do this in FFTV to substitute in mocked data.  Our object that makes Pocket network requests lives in a service locator in the Application, so we override the service locator to inject mock responses, removing a source of flakiness.

Reply:
With your comment below (to extend the actual Application object) this makes sense to me :)